modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCachePartitionExchangeManager.java

```diff
@@ -720,26 +705,68 @@ else if (customMsg instanceof WalStateAbstractMessage
         }
     }
 
-    /**
-     * @param msg Event message.
-     * @param evtType Event type.
-     */
-    private void recordEvent(String msg, int evtType) {
-        GridKernalContext ctx = cctx.kernalContext();
+    /** */
+    private void onClusterStateChangeFinish(
+        IgniteInternalFuture<AffinityTopologyVersion> fut,
+        ExchangeActions exchActions,
+        boolean baselineChanging
+    ) {
+        A.notNull(exchActions, "exchActions");
 
-        Collection<ClusterNode> srvNodes = ctx.cluster().get().forServers().nodes();
+        GridEventStorageManager evtMngr = cctx.kernalContext().event();
 
-        Collection<BaselineNode> baselineNodes = new ArrayList<>(srvNodes);
+        if (exchActions.activate() && evtMngr.isRecordable(EVT_CLUSTER_ACTIVATED) ||
+            exchActions.deactivate() && evtMngr.isRecordable(EVT_CLUSTER_DEACTIVATED) ||
+            exchActions.changedClusterState() && evtMngr.isRecordable(EVT_CLUSTER_STATE_CHANGED) ||
+            baselineChanging && evtMngr.isRecordable(EVT_BASELINE_CHANGED)
```

Comment:
why we can`t use : org.apache.ignite.internal.processors.cache.ExchangeActions#changedBaseline here ?

Reply:
Look at the method invocation.